presto-docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -259,6 +264,430 @@ Field           Required  Type      Description
+  separated value
+* :ref:`JSON encoder<json-encoder>` - Table columns are mapped to JSON
+  fields
+* :ref:`Avro encoder<avro-encoder>` - Kafka message is mapped to Avro
+  fields based on an Avro schema
+
+.. note::
+
+    A :ref:`table definition file<kafka-table-definition-files>` must be defined
+    for the encoder to work.
+
+.. _raw-encoder:
+
+raw encoder
+^^^^^^^^^^^^^^^
+
+The raw encoder formats the table columns as raw bytes using the mapping
+information specified in the
+:ref:`table definition file<kafka-table-definition-files>`.
+
+The following field attributes are supported:
+
+* ``dataFormat`` - Selects the width of the column data type
+* ``type`` - Presto data type
+* ``mapping`` - ``<start>[:<end>]``; start and end position of bytes to convert
+
+The ``dataFormat`` attribute selects the number of bytes converted.
+If absent, ``BYTE`` is assumed. All values are signed.
+
+Supported values are:
+
+* ``BYTE`` - one byte
+* ``SHORT`` - two bytes (big-endian)
+* ``INT`` - four bytes (big-endian)
+* ``LONG`` - eight bytes (big-endian)
+* ``FLOAT`` - four bytes (IEEE 754 format)
+* ``DOUBLE`` - eight bytes (IEEE 754 format)
+
+The ``type`` attribute defines the Presto data type.
+
+Depending on Presto type assigned to column different values of ``dataFormat``
+can be used:
+
+===================================== =======================================
+Presto data type                      ``dataFormat`` values
+===================================== =======================================
+``BIGINT``                            ``BYTE``, ``SHORT``, ``INT``, ``LONG``
+``INTEGER``                           ``BYTE``, ``SHORT``, ``INT``
+``SMALLINT``                          ``BYTE``, ``SHORT``
+``TINYINT``                           ``BYTE``
+``DOUBLE``                            ``DOUBLE``, ``FLOAT``
+``BOOLEAN``                           ``BYTE``, ``SHORT``, ``INT``, ``LONG``
+``VARCHAR`` / ``VARCHAR(x)``          ``BYTE``
+===================================== =======================================
+
+The ``mapping`` attribute specifies the range of bytes in a key or
+message used for encoding. It can be one or two numbers separated by a colon
+(``<start>[:<end>]``).
+
+.. note::
+
+    For ``VARCHAR`` types a start and end position must be defined. Otherwise
+    there is no way to know how many bytes the message contains. The raw format
+    mapping information is static and cannot be dynamically changed to fit the
+    variable width of some Presto data types.
+
+If only a start position is given:
+
+* For fixed width types the appropriate number of bytes are used for the
+  specified ``dateFormat`` (see above).
+
+If start and end position are given, then:
+
+* For fixed width types the size must be equal to number of bytes used by
+  specified ``dataFormat``.
+* For ``VARCHAR`` all bytes between start (inclusive) and end (exclusive) are
+  used.
+
+.. note::
```

Comment:
It's not clear what this applies to. If for varchar, it seems redundant with the above note.